IntelliTectAnalyzer/IntelliTectAnalyzer.Test/UnusedLocalVariableTests.cs

```diff
@@ -0,0 +1,154 @@
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using TestHelper;
+
+namespace IntelliTectAnalyzer.Tests
+{
+    [TestClass]
+    public class UnusedLocalVariableTests : CodeFixVerifier
+    {
+        [TestMethod]
+        public void InstanceMemberAccessedOnLocalVariable_NoDiagnosticInformationReturned()
+        {
+            string test = @"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Diagnostics;
+
+namespace ConsoleApplication1
+{
+    class TypeName
+    {   
+        public void Foo()
+        {
+            object foo = new object();
+            foo.ToString();
+        }
+    }
+}";
+
+            VerifyCSharpDiagnostic(test);
+        }
+
+        [TestMethod]
+        public void LocalVariablePassedToMethod_NoDiagnosticInformationReturned()
+        {
+            string test = @"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Diagnostics;
+
+namespace ConsoleApplication1
+{
+    class TypeName
+    {   
+        public void Foo()
+        {
+            object foo = new object();
+            Bar(foo);
+        }
+
+        public void Bar(object bar) { }
+    }
+}";
+
+            VerifyCSharpDiagnostic(test);
+        }
+
+        [TestMethod]
+        public void UnusedLocalVariable_NoDiagnosticInformationReturned()
+        {
+            string test = @"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Diagnostics;
+
+namespace ConsoleApplication1
+{
+    class TypeName
+    {   
+        public void Foo()
+        {
+            object foo = new object();
+        }
+    }
+}";
+            var expected = new DiagnosticResult
+            {
+                Id = Analyzers.UnusedLocalVariable.DiagnosticId,
+                Message = "Local variables should be used",
```

Comment:
Should we check the description here to, just for consistency?

Reply:
This `DiagnosticResult` is part of the generated code for testing analyzers. The description is actually not part of the result, and instead only apart of the descriptor for the analyzer. I will add a separate test for this.